### ISSUE TITLE
Support TS1.6 intersection types

### ISSUE THREAD

**Issue by vsiao** [palantir/tslint#788]
https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#intersection-types
```ts
interface Foo {
    foo: string;
}
interface Bar {
    bar: boolean;
}
export type t = Foo & Bar;
export var a: Array<Foo & Bar>;
```
The above test case yields the following spurious lint errors.
```
(no-bitwise) intersection.ts[7, 21]: forbidden bitwise operation
(no-bitwise) intersection.ts[8, 25]: forbidden bitwise operation
(no-unused-expression) intersection.ts[7, 21]: expected an assignment or function call
(whitespace) intersection.ts[8, 30]: missing whitespace
(whitespace) intersection.ts[8, 31]: missing whitespace
```

**Comment by JKillian** [comment]
@vsiao I'm not seeing the same issues on my end. What version of tslint are you using? Using intersection types definitely *should* be supported!

**Comment by vsiao** [comment]
oh, duh. It didn't make sense to me either! Sorry for the false alarm.
